Sources/AblyAssetTrackingInternal/WorkerQueue/Worker.swift

```diff
@@ -0,0 +1,64 @@
+import Foundation
+import AblyAssetTrackingCore
+
+/// A [Worker] interface represents workers which execute work inside [WorkerQueue].
+/// PropertiesType: - the type of properties used by this worker as both input and output.
+/// WorkerSpecificationType - the type of specification used to post another worker back to the queue.
+public protocol Worker<PropertiesType, WorkerSpecificationType>: AnyObject {
+    associatedtype PropertiesType
+    associatedtype WorkerSpecificationType
+    
+    var completion: ResultHandler<Void> { get }
```

Comment:
Yeah, on Android there's the a `CallbackWorker` abstract class. The intention here was to recreate what it does in it's overwritten methods, by creating an extension on `Worker` later on with a similar default implementation, and a call to this `completion` (callbackFunction in android).

Reply:
I don't think it makes sense for it to be on the `Worker` protocol though since it doesn't apply to all workers. Shouldn't it be a requirement of the `CallbackWorker` protocol (if that's how we choose to implement it, in line with the comment elsewhere in this PR)?

Reply:
I removed it somewhere along the way while implementing your other suggestions, 76e4ba2fa233a831d8ede80d837790a6e90ecd6c